From 52ed0096635953c229de637ed2514517f0724e35 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:14:12 -0400
Subject: [ethernet] be2net: return -ENOMEM for memory allocation failures

Message-id: <1410192881-30702-50-git-send-email-ivecera@redhat.com>
Patchwork-id: 92457
O-Subject: [RHEL7.1 PATCH 49/78] be2net: return -ENOMEM for memory allocation failures
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit 6b5686891c9858aea914c1d1c965d6bbc8a0521d
Author: Kalesh AP <kalesh.purayil@emulex.com>
Date:   Thu Jul 17 16:20:22 2014 +0530

    be2net: return -ENOMEM for memory allocation failures

    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index e0287b5..e5aad70 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -2927,7 +2927,7 @@ static int be_setup_wol(struct be_adapter *adapter, bool enable)
  cmd.va = dma_zalloc_coherent(&adapter->pdev->dev, cmd.size, &cmd.dma,
          GFP_KERNEL);
  if (cmd.va == NULL)
-  return -1;
+  return -ENOMEM;
 
  if (enable) {
   status = pci_write_config_dword(adapter->pdev,
@@ -4582,7 +4582,7 @@ static int be_stats_init(struct be_adapter *adapter)
  cmd->va = dma_zalloc_coherent(&adapter->pdev->dev, cmd->size, &cmd->dma,
           GFP_KERNEL);
  if (cmd->va == NULL)
-  return -1;
+  return -ENOMEM;
  return 0;
 }
 
-- 
1.7.1